src/main/java/com/google/devtools/build/lib/remote/ByteStreamUploader.java

```diff
@@ -323,7 +323,7 @@ private static String buildUploadResourceName(
               // level/algorithm, so we cannot know the expected committed offset
               long committedSize = committedOffset.get();
               long expected = chunker.getOffset();
-              if (!chunker.hasNext() && committedSize != expected) {
+              if (!chunker.hasNext() && committedSize != expected || (chunker.isCompressed() && committedSize == -1)) {
                 String message =
                     format(
                         "write incomplete: committed_size %d for %d total",
```

Comment:
Can you also add error message for compressed case?

Reply:
Done.